### DISCUSSION TITLE
Applying Deficiencies for Geometry Domains in Biharmonic Elasicity Problems

### DISCUSSION THREAD

**Discussion by USSENF6** [lululxvi/deepxde#1215] [General]
Dear Professor Lu:
I am working on a buckling problem for plates with stiffeners with DEEPXDE, currently, I am seeking to solve a buckling problem given the plate has an initial deflection _w*_ throughout the whole geometry domain. The PDE is defined as follows:

`def pde(x, y):
    w , F= y[:,0],y[:,1]
    dw_X = tf.gradients(w,x)[0]
    dw_x, dw_y = dw_X[:,0:1],dw_X[:,1:2]

    dw_xx = tf.gradients(dw_x, x)[0][:,0:1]
    dw_xy = tf.gradients(dw_x, x)[0][:,1:2]
    dw_yy = tf.gradients(dw_y, x)[0][:,1:2]

    dw_xxx = tf.gradients(dw_xx, x)[0][:,0:1]
    dw_yyy = tf.gradients(dw_yy, x)[0][:,1:2]

    dw_xxxx = tf.gradients(dw_xxx, x)[0][:,0:1]
    dw_yyyy = tf.gradients(dw_yyy, x)[0][:,1:2]

    dw_xxy = tf.gradients(dw_xx,x)[0][:,1:2]
    dw_xxyy = tf.gradients(dw_xxy,x)[0][:,1:2]

    dF_X = tf.gradients(w,x)[0]
    dF_x, dF_y = dF_X[:,0:1],dF_X[:,1:2]

    dF_xx = tf.gradients(dF_x, x)[0][:,0:1]
    dF_xy = tf.gradients(dF_x, x)[0][:,1:2]
    dF_yy = tf.gradients(dF_y, x)[0][:,1:2]

    dF_xxx = tf.gradients(dF_xx, x)[0][:,0:1]
    dF_yyy = tf.gradients(dF_yy, x)[0][:,1:2]

    dF_xxxx = tf.gradients(dF_xxx, x)[0][:,0:1]
    dF_yyyy = tf.gradients(dF_yyy, x)[0][:,1:2]
    
    dF_xxy = tf.gradients(dF_xx,x)[0][:,1:2]
    dF_xxyy = tf.gradients(dF_xxy,x)[0][:,1:2]
       
 
    e1 = D * (dw_yyyy + dw_xxxx + 2 * dw_xxyy) - (dw_xx * dF_yy -2 * dw_xy * dF_xy + dF_xx * dw_yy)
    e2 = (dF_yyyy + dF_xxxx + 2 * dw_xxyy) -0.5 * E * delta *(2 * dw_xx * dw_yy -  2 * dw_xy * dw_xy)
    return [e1,e2]`
In which _w_ is the deflection of the plate, and _x_ represents the coordinate of the geometry domain of a rectangular 2d plate.
E , D, and delta are global constants and are hence not in the PDE definition.

My question is that is there any methods to start the iteration of the PINN with an initial deflection value of _w=w*(x,y)_? Should it be defined in the geometry module or anywhere else?
Hope my poor English does not cause you any confusion. :)

**Comment by lululxvi** [comment]
Do you want to change the geometry during simulation? No.

You may also be interested in this paper https://arxiv.org/abs/2211.15423
